### PR TITLE
Remove usage of hass.components.hassio

### LINKED ISSUE #269: Stop accessing `hass.components.hassio`
Logger: homeassistant.helpers.frame
Source: helpers/frame.py:188
First occurred: 3:51:37 PM (2 occurrences)
Last logged: 3:51:45 PM

Detected that custom integration 'elasticsearch' accesses hass.components.hassio. This is deprecated and will stop working in Home Assistant 2024.9, it should be updated to import functions used from hassio directly at custom_components/elasticsearch/system_info.py, line 22: if self._hass.components.hassio.is_hassio()

### PR DESCRIPTION
**Pull request by legrego**
Resolves #269

### CODE REVIEW

File: custom_components/elasticsearch/system_info.py [hunk 1]
```diff
@@ -19,8 +21,8 @@ async def async_get_system_info(self):
             system_info = await self._hass.helpers.system_info.async_get_system_info()
 
             # see homeassistant/helpers/system_info.py in main home-assistant repo
-            if self._hass.components.hassio.is_hassio():
-                hostname = self._hass.components.hassio.get_host_info().get("hostname", "UNKNOWN")
+            if system_info.get("hassio"):
+                hostname = get_host_info(self._hass).get("hostname", "UNKNOWN")
```
Comment (strawgate) [inline custom_components/elasticsearch/system_info.py:25]: maybe not for this PR but we should consider not including "UNKNOWN" when the host name can't be found. perhaps just not publishing the field

Comment (legrego) [reply custom_components/elasticsearch/system_info.py]: Ah yeah thanks for the reminder. I've attempted to address in https://github.com/legrego/homeassistant-elasticsearch/pull/270/commits/d9217ab1be2f50f27ddd5a45d35e44e8f9858a21, let me know what you think.

**Comment by strawgate** [comment]
Looks good to me

**Review by strawgate — APPROVED** [review]

**Comment by strawgate** [comment]
@legrego can we merge this so I can rebase my config branch?

**Pull request merged**